[docs] Fix Tailwind CSS integration docs (#32512)
mui/material-ui: docs/data/material/guides/interoperability/interoperability.md

If you use a different framework, or already have set up your project, follow these steps:

1. Add Tailwind CSS to your project, following the instructions in https://tailwindcss.com/docs/installation.

> 2. Remove Tailwind's `base` styles in favor of the `CssBaseline` component provided by `@mui/material`, as it plays nicer with the MUI components. More details on tailwind base styles can be found [here](https://tailwindcss.com/docs/preflight#disabling-preflight)

Reviewer: Multiple things wrong on this line:

- [here] links are banned on MUI, e.g. https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/ of why
- it's not longer base that we override
- It's Tailwind CSS, not Tailwind, nor tailwind.
- We link Tailwind CSS docs without a nofollow rel link, fixed in #32514

```suggestion
2. Remove Tailwind CSS's preflight styles in favor of the `CssBaseline` component provided by `@mui/material`, as it plays nicer with the MUI components. More details on [Tailwind CSS preflight styles](https://tailwindcss.com/docs/preflight#disabling-preflight).
```

> 2. Remove Tailwind CSS's preflight style as it applies global changes that break MUI components. If you want to retrain the notion of preflight, you can use the [`CssBaseline` component](/material-ui/react-css-baseline/) that better isolates global resets. More details on [Tailwind CSS preflight styles](https://tailwindcss.com/docs/preflight).

Reviewer: **Off topic**: we could look into why it's problematic. AFAIK, with Bootstrap, we apply CSS to reset the resets of Bootstrap, we could do the same with Tailwind CSS, allowing us to use all the MUI components in an existing codebase.

**tailwind.config.js**

```diff

> module.exports = {
> +  corePlugins: {
> +    preflight: false,
> +  }
> }

Reviewer: This is how git outputs its diff, plus keeping our prettier/esling code format config:

```suggestion
 module.exports = {
+  corePlugins: {
+    preflight: false,
+  },
 };
```
